Composer/packages/client/src/components/ErrorBoundary.tsx

```diff
@@ -11,16 +11,28 @@ import { StateError } from '../recoilModel/types';
 import { ErrorPopup } from './ErrorPopup/ErrorPopup';
 
 const githubIssueUrl = `https://github.com/microsoft/BotFramework-Composer/issues`;
-const errorToShow: StateError = {
-  message: formatMessage.rich('If this problem persists, please file an issue on <a>GitHub</a>.', {
-    // eslint-disable-next-line react/display-name
-    a: ({ children }) => (
-      <a key="a" href={githubIssueUrl} rel="noopener noreferrer" style={{ color: `greenyellow` }} target="_blank">
-        {children}
-      </a>
-    ),
-  }),
-  summary: formatMessage('Something went wrong!'),
+const genericErrorTitle = 'Something went wrong!';
+
+const errorToShow = (error: any): StateError => {
+  const msg: string = typeof error === 'string' ? error : error?.message || error?.detail;
+  let messageHtml = 'If this problem persists, please file an issue on <a>GitHub</a>';
+  if (msg) {
+    messageHtml += `<details>${msg}</details>`;
+  }
+  const summary: string = msg ? msg.substring(0, 20) + '...' : genericErrorTitle;
+
+  return {
+    message: formatMessage.rich(messageHtml, {
+      // eslint-disable-next-line react/display-name
```

Comment:
Another thing I'm wondering here is, shall we define the render logic of error messages in this module `ErrorBoundary`?

Ideally, the duty of `ErrorBoundary` should be only collecting errors; it's another component `ErrorPopup` who should care about how to render an error message.

So we can move those error transforming logic to ErrorPopup.